test/hw11/test_task1.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+from All_home_works.hw11.task1 import SimplifiedEnum
+
+
+class ColorsEnum(metaclass=SimplifiedEnum):
+    RED = "RED"
+    BLUE = "BLUE"
+    ORANGE = "ORANGE"
+    BLACK = "BLACK"
+
+
+@pytest.mark.parametrize(
+    "attr",
+    [
+        "RED",
+        "BLACK",
+        "BLUE",
+        "ORANGE",
+    ],
+)
+def test_simplified_enum_metaclass_works_fine_for_class(attr):
+    assert getattr(ColorsEnum, attr) == attr
+
+
+def test_simplified_enum_metaclass__if_attr_does_not_exist():
+    with pytest.raises(AttributeError, match="Does not exist:"):
+        _ = ColorsEnum.O
+
+
+def test_simplified_enum_metaclass__if_class_instance_is_ok():
+    class Test(metaclass=SimplifiedEnum):
+        A = "A"
+
+        def __init__(self, a):
+            self.a = a
+
+        def add_one_to_a(self):
+            self.a += 1
+            return self.a
+
+    test = Test(0)
+    assert Test.A == "A"
+    assert test.A == "A"
+    assert test.a == 0
+    test.add_one_to_a()
+
+
+def test_simplified_enum_metaclass_is_ok__if_class_instance_overwrite__keys():
+    class Test(metaclass=SimplifiedEnum):
+        A = "A"
+
+        def __init__(self, a):
+            self.__keys = a
+
+    test = Test(0)
+    assert Test.A == "A"
+    assert test.A == "A"
+    assert test._Test__keys == 0
```

Comment:
What exactly you are testing here?

Reply:
That we couldn't overwrite `__keys`

Reply:
Where is the code that protects __keys?

Reply:
I didn't understand the question. But `__keys` are in metaclass.

Reply:
Why we couldn't overwrite __keys?

Reply:
Because it's a private attribute that references to metaclass.
`Test._SimplifiedEnum__keys`

Reply:
Is it the responsibility of your code to protect private variables?

Reply:
Of course, it's not

Reply:
Do we need to test this feature if you did not write it?

Reply:
Okey. I got you. It was in education purpose, will remove this test.